### DISCUSSION TITLE
Dropdown Fields - Display the Text Value

### DISCUSSION THREAD

**Discussion by st11x** [directus/directus#13761] [Archive]
Hi,

In the Collection page, the column for the dropdown field is showing the "Value". How can I make it show the "Text" value? 

For instance,

![image](https://user-images.githubusercontent.com/7398499/172241836-1b27b97f-5e42-4dc6-8a3d-1c9421a0b87a.png)

this is showing the Value as shown here

![image](https://user-images.githubusercontent.com/7398499/172241919-85454d96-9b63-4a7d-9d33-c131cf3300fb.png)

I would like it to show "Human Resource" so that it will show the correct text for user's preferred locale. I tried the Display template for the field but it didn't do anything. 

thanks
  Matt

**Comment by rijkvanzanten** [comment]
Heya! Try using the Labels display 🙂

**Reply by st11x** [reply]
What does that do? It does something but unfortunately we had that "label" in there 😛.  

![image](https://user-images.githubusercontent.com/7398499/172250680-bbe2d719-ac3b-439a-a76b-e4fd9446f078.png)

Do we have to duplicate the Choices to the Labels? I tried adding a number of things but they don't seem to do anything.

Thanks

**Reply by rijkvanzanten** [reply]
> Do we have to duplicate the Choices to the Labels? I tried adding a number of things but they don't seem to do anything.

Yup! It allows you to render any value as any formatted text. You can use the "Copy Raw Value" and "Paste Raw Value" options in the field dropdowns to easily copy/paste the configuration for the dropdown field into the display 👍🏻

**Reply by st11x** [reply]
I thought I saw the options to copy/show raw values before but I can't seem to find it in this instance. Can you guide me on how may I do that? 

That would also mean I have to copy all translations. Wished it was able to display the Text set in the Choices, it is already displaying the Value. 

Thanks
  Matt

**Reply by rijkvanzanten** [reply]
<img width="373" alt="CleanShot 2022-06-06 at 17 56 03@2x" src="https://user-images.githubusercontent.com/9141017/172255847-6782502d-c627-4ceb-8326-63b9b79e689d.png">

You can click on the field label to show the context menu 👍🏻 

There's no need to reconfigure the translated values. They're saved as a special syntax indicating that translations are used, so the same translation can be used for both configurations 🙂

**Reply by st11x** [reply]
Thanks, that was it!

**Comment by DanielRubianes** [comment]
I wrote a rough Python script utilizing httpx to accomplish this for all fields in a collection via REST routes before discovering this post...

Either way, here it is:

```
import httpx
headers = {
    "Authorization": f"Bearer {API_KEY}",
    "Content-Type": "application/json"
}

response: dict = httpx.get(f'{URL}/fields/{COLLECTION}', headers=headers).json()

original_fields = response['data']
update_fields = []

for field in original_fields:
    meta = field['meta']
    if 'options' not in meta or 'display' not in meta:
        continue
    if not meta['options'] or meta['display'] != 'labels':
        continue
    interface_options = meta['options']
    if 'choices' not in interface_options:
        continue
    field['meta']['display_options'] = interface_options
    update_fields.append(field)

for field in update_fields:
    response: dict = httpx.patch(f'{URL}/fields/{COLLECTION}', headers=headers, json=[field]).json()
    print(response)
```

This will find every field in the collection which has a specified list of choices & has display set to labels, then clone said choices over to labels.
